### PR TITLE
Implement audio quality bars

### PR DESCRIPTION
**Pull request by jfsiebel**
Incorporate audio quality bars to determine the quality of a user's network.

Requested in #5286 

This PR is an update from #6992

### CODE REVIEW

**Comment by tdjac0bs** [comment]
Hello João, as we spoke, let's improve it to have a better approach, creating something that groups  information and take decisions:

What we have:
- network connection information - (effectiveType only)

What we must add:
- add more from network connection information -  (downlink and rtt - Chrome 74);
- webrtc stats ( packetsLost, packetsSent, etc);
- reconnections (if our connection is being silently restored, we should use this information too);
- reset statistics if network changes (if user disconnect from one network and connect again, we should reset statistics changing state to gathering data);

PS:
- States can be: gathering, good, bad
- We'll probably have one webrtc stats for each peer connection (check if they show different values and consider it);
- We need to simulate what are the values that affects the quality (by simulating high latency,uplink saturation, downlink saturation, etc);

Ideas for Future:
- We can make the client connect one video after the other, allowing to collect statistics and keep history about the number of streams (webcams), like:
 --- 1 stream - good
 --- 2 streams - good
 --- 3 streams - bad (suggest user to switch to "active talker" video mode);

**Comment by capilkey** [comment]
Would this be only taking into account the metrics available to the client? There are things we can get from FreeSWITCH about what it thinks the audio information being sent to it.

File: bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  const selector = {
+    userId,
+  };
+
+  const modifier = {
+    $set: {},
+  };
+
+  modifier.$set[setting] = value;
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js]: Can't we use `[setting]` in the modifier above? Maybe

```
$set: {`${setting}`:value}
```

File: bigbluebutton-html5/imports/ui/components/video-provider/component.jsx [hunk 2]
```diff
@@ -158,6 +165,28 @@ class VideoProvider extends Component {
 
     this.visibility.onVisible(this.unpauseViewers);
     this.visibility.onHidden(this.pauseViewers);
+
+    this.currentWebcamsStatsInterval = setInterval(() => {
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/video-provider/component.jsx]: Do you think we should be able to opt-out of re-running this every 5s on a device with good connection but low cpu? Or we assume that if there are webcams visible, then we should be able to re-run it

File: bigbluebutton-html5/imports/ui/components/slow-connection/styles.scss [hunk 3]
```diff
@@ -0,0 +1,34 @@
+.sizingBox {
+  width: 27px;
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/slow-connection/styles.scss]: magic numbers warning

File: bigbluebutton-html5/imports/ui/components/user-list/user-list-content/user-participants/user-list-item/user-icons/component.jsx [hunk 4]
```diff
@@ -1,41 +1,50 @@
...
-          : null
+        ) : null
       }
     </div>
   );
 };
 
 UserIcons.propTypes = propTypes;
-export default UserIcons;
+export default memo(UserIcons);
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/user-list/user-list-content/user-participants/user-list-item/user-icons/component.jsx]: :clap: https://scotch.io/tutorials/react-166-reactmemo-for-functional-components-rendering-control#toc-what-is-react-memo- using react memo

File: bigbluebutton-html5/imports/api/users/server/methods/setUserEffectiveConnectionType.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import { Meteor } from 'meteor/meteor';
+import { check } from 'meteor/check';
+import RedisPubSub from '/imports/startup/server/redis';
+import Logger from '/imports/startup/server/logger';
+import setEffectiveConnectionType from '../modifiers/setUserEffectiveConnectionType';
+
+export default function setUserEffectiveConnectionType(credentials, effectiveConnectionType) {
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/users/server/methods/setUserEffectiveConnectionType.js]: Perhaps check if settings.yml has `enableNetworkInfo`  enabled before proceeding with method calls

File: bigbluebutton-html5/imports/api/users/server/methods/userChangedSettings.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  const CHANNEL = REDIS_CONFIG.channels.toAkkaApps;
+  const EVENT_NAME = 'UserChangedSettingsEvtMsg';
+
+  const { meetingId, requesterUserId } = credentials;
+
+  if (!meetingId || !requesterUserId) return;
+
+  check(meetingId, String);
+  check(requesterUserId, String);
+  check(setting, String);
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/users/server/methods/userChangedSettings.js]: What is the type of 'value'?

File: bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import { check } from 'meteor/check';
+import Users from '/imports/api/users';
+import Logger from '/imports/startup/server/logger';
+
+export default function setChangedSettings(userId, setting, value) {
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js]: What settings are we referring to here - perhaps we can improve the function name

File: bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  check(userId, String);
+  check(setting, Match.Maybe(String));
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js]: `setting` was checked as a String in  bigbluebutton-html5/imports/api/users/server/methods/userChangedSettings.js 

Why would it be a Maybe (String) here? We likely don't need to check it again either way

File: bigbluebutton-html5/imports/api/users/server/methods/userChangedSettings.js (hunk 6 above)
Comment (jfsiebel) [reply bigbluebutton-html5/imports/api/users/server/methods/userChangedSettings.js]: The type of `value` could be any value, update with `Match.Any`.

File: bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js (hunk 7 above)
Comment (jfsiebel) [reply bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js]: I'm referring to any settings changed, what name do you suggest?

File: bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js (hunk 8 above)
Comment (jfsiebel) [reply bigbluebutton-html5/imports/api/users/server/modifiers/setChangedSettings.js]: You are right, updated.

**Pull request merged**